src/main/java/thePackmaster/patches/bitingcoldpack/OnApplyDebuffPatch.java

```diff
@@ -0,0 +1,44 @@
+package thePackmaster.patches.bitingcoldpack;
+
+import com.evacipated.cardcrawl.modthespire.lib.*;
+import com.megacrit.cardcrawl.actions.common.ApplyPowerAction;
+import com.megacrit.cardcrawl.actions.utility.DiscardToHandAction;
+import com.megacrit.cardcrawl.cards.AbstractCard;
+import com.megacrit.cardcrawl.core.AbstractCreature;
+import com.megacrit.cardcrawl.dungeons.AbstractDungeon;
+import com.megacrit.cardcrawl.powers.AbstractPower;
+import javassist.CtBehavior;
+import thePackmaster.actions.bitingcoldpack.ActivateGlaciateAction;
+import thePackmaster.cards.bitingcoldpack.GrowingAffliction;
+import thePackmaster.powers.bitingcoldpack.FrostbitePower;
+import thePackmaster.powers.bitingcoldpack.GlaciatePower;
+
+import java.lang.reflect.Field;
+
+// Patch for effects that activate upon the player applying a debuff
+@SpirePatch(clz = ApplyPowerAction.class, method = "update")
```

Comment:
This could be done in `receivePostPowerApplySubscriber` in SpireAnniversary5Mod
That way we don't reinvent the wheel for the same thing multiple times.